apps/multiplatform/common/src/commonMain/kotlin/chat/simplex/common/views/chat/ChatView.kt

```diff
@@ -1497,6 +1541,78 @@ private fun TopEndFloatingButton(
   }
 }
 
+@Composable
+private fun TopCenterFloatingButton(
+  modifier: Modifier,
+  lastVisibleItem: ChatItem?,
+  isNearBottom: Boolean,
+  firstVisibleItemScrollOffset: Int
+) {
+  val coroutineScope = rememberCoroutineScope()
+  var hideDateWhenNotScrolling: Job? by remember { mutableStateOf(null) }
+  val firstVisibleItemDate = remember { mutableStateOf<Instant?>(null) }
+
+
+  val showDate = remember { mutableStateOf(false) }
+  fun setDateVisibility(isVisible: Boolean) {
+    if (isVisible) {
+      val date = firstVisibleItemDate.value
+      val now = Clock.System.now()
+      if (!isNearBottom && !showDate.value && date != null && getTimestampDateText(date) != getTimestampDateText(now)) {
+        showDate.value = true
+      }
+    } else if (showDate.value) {
+      showDate.value = false
+    }
+  }
+
+  KeyChangeEffect(firstVisibleItemScrollOffset) {
+    hideDateWhenNotScrolling?.cancel()
+
+    hideDateWhenNotScrolling = coroutineScope.launch {
+      delay(1000)
```

Comment:
If you need a delay, why not to add it before `hideDateWhenNotScrolling = coroutineScope.launch {`?
Because if the coroutine gets canceled before reaching the delay, it still try to display date which is not needed at that point of time, probably

Reply:
I was wrong, I understood your idea incorrectly. Reverted your commit to return that old behaviour and simplified it a little here: 
https://github.com/simplex-chat/simplex-chat/pull/4951/commits/b213e34df0c324d578f63cee53951393cafd1d33